cmd/kubeadm/app/cmd/config.go

```diff
@@ -571,11 +585,45 @@ type ImagesList struct {
 	cfg *kubeadmapi.InitConfiguration
 }
 
+// imageTextPrinter prints image info in a text form
+type imageTextPrinter struct {
+	output.TextPrinter
+}
+
+// PrintObj is an implementation of ResourcePrinter.PrintObj for plain text output
+func (itp *imageTextPrinter) PrintObj(obj runtime.Object, writer io.Writer) error {
+	for _, img := range obj.(*outputapiv1alpha1.Images).Images {
+		if _, err := fmt.Fprintf(writer, "%s/%s:%s\n", img.Prefix, img.Name, img.Tag); err != nil {
+			return err
+		}
+	}
+	return nil
+}
+
+// imageTextPrintFlags provides flags necessary for printing image in a text form.
+type imageTextPrintFlags struct{}
+
+// ToPrinter returns kubeadm printer for the text output format
+func (ipf *imageTextPrintFlags) ToPrinter(outputFormat string) (output.Printer, error) {
+	if outputFormat == output.TextOutput {
+		return &imageTextPrinter{}, nil
+	}
+	return nil, genericclioptions.NoCompatiblePrinterError{OutputFormat: &outputFormat, AllowedFormats: []string{output.TextOutput}}
+}
+
 // Run runs the images command and writes the result to the io.Writer passed in
-func (i *ImagesList) Run(out io.Writer) error {
+func (i *ImagesList) Run(out io.Writer, printer output.Printer) error {
 	imgs := images.GetControlPlaneImages(&i.cfg.ClusterConfiguration)
+
+	var outImages []outputapiv1alpha1.Image
 	for _, img := range imgs {
-		fmt.Fprintln(out, img)
+		splitted := strings.SplitN(img, "/", 2)
+		nameTag := strings.SplitN(splitted[1], ":", 2)
```

Comment:
[1]

Reply:
i guess @rosti's comment still applies here:
https://github.com/kubernetes/kubernetes/pull/86810#discussion_r363297497

can we use `@` to determine if this is a digest or name+tag?

then again our structure field is called `tag`, does it make sense to store digests in there at some point?

Reply:
This parsing based on [this formatting](https://github.com/kubernetes/kubernetes/blob/release-1.17/cmd/kubeadm/app/images/images.go#L32). "tag" and "prefix" names come from there.
if we want change that we'll have to do it in both places.

Reply:
let's keep it out of scope of this PR then.
@rosti WDYT?

Reply:
no splitting image name to components in the current version. Please review.